stdlib/std.ncl

```diff
@@ -2656,4 +2641,18 @@
       ```
     "%
     = fun msg => null | FailWith msg,
+
+  merge_all
+    : Array Dyn -> Dyn
+    | doc m%"
+      Merges an array of records.
```

Comment:
```suggestion
      Merges an array of values.
```

Reply:
Would that make sense to restrict the type further to `{_ : Dyn}`? On one hand, there's not many value that are interesting to merge beside record (otherwise it's going to fail unless the operands are equals). On the other hand merge does accept to merge more stuff than records, although it is _very_ restricted.

Reply:
That's sort of what we had before. I felt that it doesn't make much sense to restrict `merge_all` if we don't restrict `(&)` to records. Especially if we end up pointwise merging of arrays or something like that in the future.